zerver/views/messages.py

```diff
@@ -999,10 +1000,13 @@ def send_message_backend(request: HttpRequest, user_profile: UserProfile,
     else:
         sender = user_profile
 
-    if defer_until:
+    if (delivery_type == 'send_later' or delivery_type == 'remind') and defer_until is None:
+        return json_error(_("Missing deliver_at in a request for delayed message delivery"))
```

Comment:
We should probably be consistent on saying either `defer_until` or `deliver_at`.  Both names seem fine to me.